[refactor] 캘린더 수정

changyeopbaek opened the pull request:
## PR 유형

- [ ] 새로운 기능 추가
- [ ] 버그 수정
- [x] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항 (오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [x] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정/삭제

---

## PR 체크리스트

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다. (버그 수정/기능에 대한 테스트)

---

### PR 상세

---

## 이슈

resolves #

changyeopbaek committed:
[refactor] 캘린더 기능 개선

- 이미 선택된 날짜를 다시 해세 가능
- 날짜 선택 해제 시 전체 대회 리스트로 복귀하도록 구현
- 킴보드로 날짜 셀 접근 및 선택/해제 가능

changyeopbaek committed:
[fix] 빌드 에러 수정

- 빌드 에러로 인해 node_modules  삭제 후 재설치

rhocci commented on src/components/MonthlyCalendar/MonthlyCalendar.css.ts:
```diff
@@ -95,9 +95,18 @@ export const cell = style({
+      outline: 'none',
+    },
+    '&:focus-visible': {
+      outline: `2px solid ${tokens.color.field.border.focus}`,
```
이거 포커스 스타일 글로벌로 들어가있어서 이 한줄은 삭제해도 될거같아유

rhocci commented on src/components/MonthlyCalendar/MonthlyCalendar.tsx:
```diff
@@ -144,24 +144,43 @@ export function MonthlyCalendar({
+                    } else {
+                      setDate(date);
+                    }
+                  }
```
[위에서 코파일럿이 리뷰한 부분](https://github.com/FRONTENDBOOTCAMP-14th/shuttlers.gg/pull/92/files#r2503669062)이랑 중복 코드라고 하는데 둘 중 하나만 쓰던지 코드를 통합하셔야 할 거 같아요!

changyeopbaek committed:
[refactor] 캘린더 날자 선택 /해제 토글 로직 중복 제거 및 접근성 개선

- 나랒 셀의 oncLick, onkeyDown에서 중복되던 선택/해제 로직을 handleDateToggle 함수로 분리
- 마우스 클릭과 키보드(Enter/Space) 모두 동일한 토글 동작 구현
- 코드 가독성 및 유지보수성 개선

changyeopbaek committed:
[refactor] CSS 수정

- MonthlyCalendar.css 내 cell 클릭 시 focus style 수정

rhocci committed:
[style] 중복구문 삭제
